templates/hardhat/solidity/scripts/interact.ts

```diff
@@ -0,0 +1,37 @@
+// Script that interacts with a Greeter contract
+
+// Address of the contract to interact with
+const CONTRACT_ADDRESS = "";
+if (!CONTRACT_ADDRESS) throw "⛔️ Provide address of the contract to interact with!";
+
+async function main() {
+  console.log(`Running script to interact with contract ${CONTRACT_ADDRESS}`);
+
+  // Get the first signer
+  const [signer] = await hre.ethers.getSigners();
+  
+  // Get the contract factory and deploy
+  const Greeter = await hre.ethers.getContractFactory("Greeter");
+  const greeterContract = await Greeter.connect(signer).attach(CONTRACT_ADDRESS);
```

Comment:
```suggestion
  const greeterContract = Greeter.connect(signer).attach(CONTRACT_ADDRESS);
```

Adding the `ethers` import (& type analysis) shows that the `await` here isn't needed